src/thor/bidirectional_astar.cc

```diff
@@ -1016,6 +1040,10 @@ std::vector<std::vector<PathInfo>> BidirectionalAStar::FormPath(GraphReader& gra
       const BDEdgeLabel& edgelabel = edgelabels_forward_[edgelabel_index];
 
       const auto* edge = graphreader.directededge(edgelabel.edgeid(), tile);
+      if (edge == nullptr) {
+        throw std::runtime_error(
+            "tile cache is connected to tile source via a piece of wet spaghetti and is randomly able and unable to load tiles. please ask the implementer to think a bit more clearly about how he designs a cache");
```

Comment:
i didnt mean you should literally use that string as the exception... my point is broader, instead of blindly continuing when you cant find a tile you need to t hink first, should have i been able to get this tile? is the algorithm at a point where it has clearly seen this tile before but now for some reason cannot get the tile again. in these cases we need to throw immediately and stop the computation.

Reply:
:D 
Sorry about that, I will think about a normal message.